libs/wingc/src/type_check.rs

```diff
@@ -1106,6 +1130,12 @@ impl<'a> TypeChecker<'a> {
 
 				container_type
 			}
+			ExprKind::JsonLiteral { type_, element } => {
+				*self.is_json.borrow_mut() = true;
+				let json_type = self.resolve_type_annotation(type_, env, statement_idx);
```

Comment:
I think the `type_` here is always either `Json` or `MutJson` so there's no need to `resolve_type..` here.
this should ways return `self.types.[mut_]json()`.
What is missing here is some recursion into the literals elements and verifying they are of a valid json type.